compiler/rustc_privacy/src/lib.rs

```diff
@@ -1311,9 +1311,8 @@ impl<'tcx> Visitor<'tcx> for TypePrivacyVisitor<'tcx> {
                 let sess = self.tcx.sess;
                 let sm = sess.source_map();
                 let name = match qpath {
-                    hir::QPath::Resolved(..) | hir::QPath::LangItem(..) => {
-                        sm.span_to_snippet(qpath.span()).ok()
-                    }
+                    hir::QPath::LangItem(..) => sm.span_to_snippet(qpath.span()).ok(),
```

Comment:
Don't lang items have a string representation you can take here? It just feels a bit silly to use span_to_snippet for this (but not your circus, not this PR)